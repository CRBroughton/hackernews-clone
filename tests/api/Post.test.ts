// tests/Post.test.ts
import { createTestContext } from '../__helpers'
const ctx = createTestContext()
it('Queries the database feed', async() => {
  const feed = await ctx.client.request(`           
    query {
      feed {           
        id
        description
        url
        postedBy {
          id
          name
        }
        voters {
          id
        }
      }
    }
  `)
  expect(feed).toMatchSnapshot()
  const persistedData = await ctx.prisma.post.findMany()
  expect(persistedData).toMatchSnapshot()
})
