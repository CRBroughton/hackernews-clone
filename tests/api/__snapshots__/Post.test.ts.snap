// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries the database feed 1`] = `
Object {
  "feed": Array [
    Object {
      "description": "Vite.js",
      "id": "040688e1-c08f-4752-8d5f-888ffb36ef02",
      "postedBy": Object {
        "id": "97527cbe-2de0-40cd-a953-8caae780e66e",
        "name": "Alice",
      },
      "url": "https://vitejs.dev",
      "voters": Array [],
    },
    Object {
      "description": "My Portfolio",
      "id": "2df7837b-788d-4410-aa65-919da54f7175",
      "postedBy": Object {
        "id": "97527cbe-2de0-40cd-a953-8caae780e66e",
        "name": "Alice",
      },
      "url": "https://crbroughton.me",
      "voters": Array [],
    },
    Object {
      "description": "Code-First GraphQL schemas for JavaScript/TypeScript",
      "id": "16e11628-462a-4235-afb2-21c0db7425ba",
      "postedBy": Object {
        "id": "97527cbe-2de0-40cd-a953-8caae780e66e",
        "name": "Alice",
      },
      "url": "https://nexusjs.org",
      "voters": Array [],
    },
  ],
}
`;

exports[`Queries the database feed 2`] = `
Array [
  Object {
    "createdAt": 2022-02-15T20:24:21.706Z,
    "description": "Code-First GraphQL schemas for JavaScript/TypeScript",
    "id": "16e11628-462a-4235-afb2-21c0db7425ba",
    "postedById": "97527cbe-2de0-40cd-a953-8caae780e66e",
    "topic": "General",
    "url": "https://nexusjs.org",
  },
  Object {
    "createdAt": 2022-02-15T20:24:21.706Z,
    "description": "My Portfolio",
    "id": "2df7837b-788d-4410-aa65-919da54f7175",
    "postedById": "97527cbe-2de0-40cd-a953-8caae780e66e",
    "topic": "General",
    "url": "https://crbroughton.me",
  },
  Object {
    "createdAt": 2022-02-15T20:24:21.706Z,
    "description": "Vite.js",
    "id": "040688e1-c08f-4752-8d5f-888ffb36ef02",
    "postedById": "97527cbe-2de0-40cd-a953-8caae780e66e",
    "topic": "General",
    "url": "https://vitejs.dev",
  },
]
`;
